caikit_nlp/modules/text_generation/fine_tuning.py

```diff
@@ -247,13 +228,27 @@ def run(
             # and thus the device placement be according to training strategy,
             # its better to let Trainer handle the evaluation / prediction
 
-            # TODO: Add support for passing extra arguments to prediction_step
+            generate_args = {
+                "prediction_loss_only": False,
+            }
+            if isinstance(self.model, Seq2SeqTrainer):
+                generate_args["max_new_tokens"] = max_new_tokens
+                generate_args["min_new_tokens"] = min_new_tokens
+            else:
+                # NOTE: Currently the default trainer doesn't support easy way to run individual
+                # samples without converting them into Datasets etc. There is a
+                # predict_with_generate flag, but it doesn't do anything.
+                # Applicable for transformers==4.31.0
+                error(
+                    "<NLP39984681E>",
+                    NotImplementedError(
+                        f"Generation on {type(self.model)} not support \
+                      currently! Please try saving and running this model in TGIS."
```

Comment:
oof. Does exporting via the trainer save API + reloading give you a transformer model back? I wonder if it would be better to have the first inference call export and reload with a warning until we find something better / implement a causal LM trainer doing something similar. Slow feels better than completely broken here IMO.

Or, is there any way we can cast to the seq2seq trainer and leverage the generate API for that? I guess that probably doesn't handle shifting etc the same way...

Reply:
yeah, I think converting the seq2seq could land with weird mismatch issues. 

Saving and reloading is certainly an option. It would simplify this block of `run` function entirely. But could be more inefficient, since the model is already on appropriate devices at this point, so loading them again, we would loose the distribution, which is mainly what I was trying to persist here. 

But certainly, not having a solution of causal lm would not be great.